### DISCUSSION TITLE
Swap current with largest active window

### DISCUSSION THREAD

**Discussion by fiskhest** [hyprwm/Hyprland#5060] [General]
Apologies if I'm overlooking something obvious;
Using the dwindle-layout, I want to put focus on one window and press a keybind executing a script/dispatcher that will swap its position with the largest window in the current workspace. If the focused window is the largest window, swap it with the previously active window. iiuc the current dispatcher terminology correctly, something like `swaplargestorpreviousactive`

Took a look at the currently documented dispatchers but was unable to figure out a clever solution without relying on a lot of janky scripting. Maybe this is something more easily resolved upstream?

**Comment by vaxerski** [comment]
script or plugin

**Reply by fiskhest** [reply]
While not quite exactly what I was imagining, the new [layoutmsg](https://github.com/hyprwm/Hyprland/pull/7903) `movetoroot` is a step in the right direction.
